### ISSUE TITLE
Unexpected Azure Dependency with OpenAI and Langchain Library in Next.js App Deployed on AWS Amplify

### ISSUE THREAD

**Issue by vrunm** [langchain-ai/langchain#8877]
### System Info

langchain 0.0.256 
Python 3.8

### Who can help?

@hwchase17 

### Information

- [ ] The official example notebooks/scripts
- [x] My own modified scripts

### Related Components

- [X] LLMs/Chat Models
- [ ] Embedding Models
- [X] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [X] Document Loaders
- [X] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

I've developed a Next.js application that uses the Langchain library for chat functionality and is deployed on AWS Amplify. The application works perfectly when running locally, but fails after deployment on AWS Amplify.

The application uses the Langchain library, OpenAIEmbeddings for generating embeddings, and PineconeStore for storing vectors. I've ensured that my environment variables are correctly set up both locally and in the AWS Amplify console.

Here is the code snippet for my chat handler:

```
import type { NextApiRequest, NextApiResponse } from 'next';
import { OpenAIEmbeddings } from 'langchain/embeddings/openai';
import { PineconeStore } from 'langchain/vectorstores/pinecone';
import { makeChain } from '@/utils/makechain';
import { pinecone } from '@/utils/pinecone-client';
import { PINECONE_INDEX_NAME, PINECONE_NAME_SPACE } from '@/config/pinecone';

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse,
) {
  const { question, history } = req.body;

  if (req.method !== 'POST') {
    res.status(405).json({ error: 'Method not allowed' });
    return;
  }

  if (!question) {
    return res.status(400).json({ message: 'No question in the request' });
  }

  const sanitizedQuestion = question.trim().replaceAll('\n', ' ');

  try {
    const index = pinecone.Index(PINECONE_INDEX_NAME);
    const vectorStore = await PineconeStore.fromExistingIndex(
      new OpenAIEmbeddings(),
      {
        pineconeIndex: index,
        textKey: 'text',
        namespace: PINECONE_NAME_SPACE,
      },
    );

    const chain = makeChain(vectorStore);
    const response = await chain.call({
      question: sanitizedQuestion,
      chat_history: history || [],
    });

    res.status(200).json(response);
  } catch (error: any) {
    console.log('chat.ts file error: ', error);
    res.status(500).json({ error: error.message || 'Something went wrong' });
  }
}
```

### Expected behavior

The error message I'm receiving suggests that the Langchain library is assuming I am running on an Azure instance and is expecting an Azure-specific environment variable. However, I am not using Azure, I'm using AWS. The error message states that an 'azureOpenAIApiInstanceName' is missing, which, as I understand, is only relevant if I was using Azure.

Has anyone encountered a similar issue, or have any insights into why this might be happening? I've been unable to find any information in the Langchain library documentation about this Azure dependency.

**Comment by vrunm** [comment]
The error message states that an 'azureOpenAIApiInstanceName' is missing, which, as I understand, is only relevant if I was using Azure. I am currently suing the AWS instance so the above checks to do for setting the azure variables is not relevant.

> The error message you're seeing suggests that the library is trying to authenticate with Azure's Cognitive Services, but it's unable to find the necessary environment variables. 
This is not the correct interpretation of the error I am encountering. It is not authenticating the Azure cognitive services. The issue lies in the error message being generated when I am using AWS instance and not Azure.

**Comment by vrunm** [comment]
I managed to resolve the issue by ensuring that the necessary environment variables were passed into the `.env.production` file during the build phase on AWS Amplify. The solution was to modify the `amplify.yml ` build specification to include these environment variables.

Here's a brief explanation for anyone encountering a similar problem:

By default, a Next.js server component doesn't have access to environment variables set in AWS Amplify. So, even if you correctly set up the environment variables in the AWS Amplify Console, you'll need to ensure they're accessible at runtime for your Next.js application.

Modify the amplify.yml file:
```
version: 1
frontend:
  phases:
    preBuild:
      commands:
        - npm ci
    build:
      commands:
        - env | grep -e OPENAI_API_TYPE -e YOUR_OTHER_ENV_VAR >> .env.production
        - npm run build
  artifacts:
    baseDirectory: .next
    files:
      - '**/*'
  cache:
    paths:
      - node_modules/**/*
      - .next/cache/**/*
```
Replace `YOUR_OTHER_ENV_VAR` with any other environment variables you might need.

Deploy & Test: After updating the amplify.yml file, I pushed the changes, triggering a new build and deployment process on AWS Amplify. After deployment, the application accessed the environment variables correctly at runtime, and the error was resolved.

In conclusion, the key takeaway is ensuring that any environment variables required during runtime are added to the `.env.production` file in the build phase on AWS Amplify. This makes them accessible to your Next.js server components when the application is live.

**Comment by Santy1422** [comment]
Hi have this problem 
my amplify.yml

version: 1
frontend:
  phases:
    preBuild:
      
     
commands:
        - npm ci
        - echo "SPORTMONKS_API_TOKEN=$SPORTMONKS_API_TOKEN" >> .env.production
        - echo "MONGODB_URI=$MONGODB_URI" >> .env.production
        - echo "STRIPE_SECRET_KEY=$STRIPE_SECRET_KEY" >> .env.production
        - echo "STRIPE_WEBHOOK_SECRET=$STRIPE_WEBHOOK_SECRET" >> .env.production
        - echo "NEXT_PUBLIC_STRIPE_PUBLISHABLE_KEY=$NEXT_PUBLIC_STRIPE_PUBLISHABLE_KEY" >> .env.production
        - echo "JWT_SECRET_PASSWORD=$JWT_SECRET_PASSWORD" >> .env.production
        - echo "OPENAI_API_TYPE =$OPENAI_API_KEY" >> .env.production

       
- echo "OPENAI_API_KEY=$OPENAI_API_KEY" >> .env.production
        - echo "PINECONE_API_KEY=$PINECONE_API_KEY" >> .env.production
        
       
- echo "PINECONE_ENVIRONMENT=$PINECONE_ENVIRONMENT" >> .env.production
        - echo "PINECONE_INDEX_NAME=$PINECONE_INDEX_NAME" >> .env.production
    build:
      commands:
        - npm run build
  artifacts:
    baseDirectory: .next
    files:
      
     
- '**/*'
  cache:
    paths:
      - node_modules/**/*
      - .next/cache/**/*
